explainability-api/microservice/checklist.py

```diff
@@ -0,0 +1,179 @@
+# add functionality to process json to be injected into the db
+import json
+import logging
+import requests
+import itertools
+
+from typing import List
+
+
+def create_query(skill, test_cases: List):
+    """
+    Creates a query and make it suitable for sending to for prediction
+
+    Args:
+        skill: input skill for which the checklist tests are run
+        test_cases (list) : Test cases as a list
+
+    Returns:
+        json_object (json object) : A json object containing the test case and its prediction
+        answer (str) : Prediction for test case made by the skill
+
+    """
+    skill_type = skill["skill_type"]
+    base_model = skill["default_skill_args"].get("base_model")
+    adapter = skill["default_skill_args"].get("adapter")
+    # extract all tests
+    all_tests = [tests["test_cases"] for tests in test_cases]
+    # all_tests = list(itertools.chain.from_iterable([tests["test_cases"] for tests in test_cases]))
+    questions, contexts, answers = list(), list(), list()
+
+    test_type = list(itertools.chain.from_iterable([[test["test_type"]] * len(test["test_cases"])
```

Comment:
I also like list comprehension in python, but here you loop three times over the same list (`test_cases`) which can be done in a single loop.

Reply:
Thank you! I sucked right there :D